exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -589,8 +589,59 @@ func (p *Poloniex) CancelAllOrders(_ *order.Cancel) (order.CancelAllResponse, er
 
 // GetOrderInfo returns order information based on order ID
 func (p *Poloniex) GetOrderInfo(orderID string, pair currency.Pair, assetType asset.Item) (order.Detail, error) {
-	var orderDetail order.Detail
-	return orderDetail, common.ErrNotYetImplemented
+	orderInfo := order.Detail{
+		Exchange: p.Name,
+		Pair:     pair,
+	}
+
+	trades, err := p.GetAuthenticatedOrderTrades(orderID)
+	if err == nil {
+		for _, o := range trades {
+			var tradeHistory order.TradeHistory
+			tradeHistory.Exchange = p.Name
+			tradeHistory.Side, err = order.StringToOrderSide(o.Type)
+			if err != nil {
+				return orderInfo, err
+			}
+			tradeHistory.TID = strconv.FormatInt(o.GlobalTradeID, 10)
+			tradeHistory.Timestamp, err = time.Parse(common.SimpleTimeFormat, o.Date)
+			if err != nil {
+				return orderInfo, err
+			}
+			tradeHistory.Price = o.Rate
+			tradeHistory.Amount = o.Amount
+			tradeHistory.Total = o.Total
+			tradeHistory.Fee = o.Fee
+			orderInfo.Trades = append(orderInfo.Trades, tradeHistory)
+		}
+	}
+
+	resp, err := p.GetAuthenticatedOrderStatus(orderID)
+	if err != nil {
+		if len(orderInfo.Trades) > 0 { // on closed orders return trades only
```

Comment:
Poloniex has a really strange API for this https://docs.poloniex.com/#returnorderstatus so unfortunately this logic cannot be changed.
Does the error contain the status of the order? Unfortunately I don't have anything that can get to this scenario. It would be helpful to set the status of the returned orderInfo so at least the user can be made aware of why they are not receiving their requested order information.

Reply:
OK, we can set manually order status as CLOSED if GetAuthenticatedOrderStatus return  "Order not found" while we have trades for this order.